daemon-tests/src/mocks/oracle.rs

```diff
@@ -30,41 +31,53 @@ impl xtra::Actor for OracleActor {
 
     async fn stopped(self) -> Self::Stop {}
 }
-impl Oracle for OracleActor {}
 
 #[xtra_productivity(message_impl = false)]
 impl OracleActor {
     async fn handle(
         &mut self,
-        msg: oracle::GetAnnouncement,
+        _msg: oracle::GetAnnouncement,
     ) -> Result<olivia::Announcement, oracle::NoAnnouncement> {
-        self.mock.lock().await.get_announcement(msg)
+        let announcement = self
+            .mock
+            .lock()
+            .await
+            .announcement
+            .clone()
+            .expect("announcement has been set");
```

Comment:
Why panic when we can return an error?

Reply:
I thought i wouldn't matter here since it is a test but i guess its better to return the error since we already have an error return type